From ccf9a34993e9ad2411c538498227bbc789d41bb6 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:13:08 -0400
Subject: [ethernet] i40evf: Remove DEFINE_PCI_DEVICE_TABLE macro use

Message-id: <1409663591-32531-146-git-send-email-sassmann@redhat.com>
Patchwork-id: 90621
O-Subject: [RHEL7.1 PATCH 145/148] PCI: Remove DEFINE_PCI_DEVICE_TABLE macro use
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Benoit Taine <benoit.taine@lip6.fr>

We should prefer `struct pci_device_id` over `DEFINE_PCI_DEVICE_TABLE` to
meet kernel coding style guidelines.  This issue was reported by checkpatch.

A simplified version of the semantic patch that makes this change is as
follows (http://coccinelle.lip6.fr/):

// <smpl>

@@
identifier i;
declarer name DEFINE_PCI_DEVICE_TABLE;
initializer z;
@@

- DEFINE_PCI_DEVICE_TABLE(i)
+ const struct pci_device_id i[]
= z;

// </smpl>

[bhelgaas: add semantic patch]
Signed-off-by: Benoit Taine <benoit.taine@lip6.fr>
Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 9baa3c34ac4e27f7e062f266f50cc5dbea26a6c1)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index ab15f4d..38429fa 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -49,7 +49,7 @@ static const char i40evf_copyright[] =
  * { Vendor ID, Device ID, SubVendor ID, SubDevice ID,
  *   Class, Class Mask, private data (not used) }
  */
-static DEFINE_PCI_DEVICE_TABLE(i40evf_pci_tbl) = {
+static const struct pci_device_id i40evf_pci_tbl[] = {
  {PCI_VDEVICE(INTEL, I40E_DEV_ID_VF), 0},
  /* required last entry */
  {0, }
-- 
1.7.1